File: sample_apps/sdksampleapps/typescript/kb_example/src/logger.ts
/**
 * Standard logger for kb_example scripts.
 * Use this instead of console so output is consistent and easy to follow.
 */

const LEVEL = {
  INFO: "info",
  WARN: "warn",
  ERROR: "error",
} as const;

function formatMessage(level: string, message: string, rest: unknown[]): void {
  const prefix = `[${level.toUpperCase()}]`;
  const args = rest.length > 0 ? rest : [];
  switch (level) {
    case LEVEL.WARN:
      console.warn(prefix, message, ...args);
      break;
    case LEVEL.ERROR:
      console.error(prefix, message, ...args);
      break;
    default:
      console.log(prefix, message, ...args);
  }
}

export const logger = {
  /** Log an informational message. */
  info(message: string, ...args: unknown[]): void {
    formatMessage(LEVEL.INFO, message, args);
  },

  /** Log a warning. */
  warn(message: string, ...args: unknown[]): void {
    formatMessage(LEVEL.WARN, message, args);
  },

  /** Log an error. */
  error(message: string, ...args: unknown[]): void {
    formatMessage(LEVEL.ERROR, message, args);
  },

  /** Log a label and a value as pretty-printed JSON. Use for API responses. */
  json(label: string, value: unknown): void {
    const formatted =
      typeof value === "string" ? value : JSON.stringify(value, null, 2);
    console.log(`[INFO] ${label}\n${formatted}`);
  },
};

File: sample_apps/sdksampleapps/python/agent_example/src/logger.py
"""
Shared logging for the agent_example sample.

Simple console logger: prefixes lines with [INFO], [WARN], [ERROR].
Use logger.json(label, value) to pretty-print API responses or objects for debugging.
"""
import json


def _format_message(level: str, message: str, *args: object) -> None:
    """Print a single line prefixed by [LEVEL]."""
    prefix = f"[{level.upper()}]"
    if args:
        print(prefix, message, *args)
    else:
        print(prefix, message)


class _Logger:
    """Minimal logger used by all examples and src/index.py."""

    @staticmethod
    def info(message: str, *args: object) -> None:
        """Log an informational message. Extra args are printed after the message."""
        _format_message("INFO", message, *args)

    @staticmethod
    def warn(message: str, *args: object) -> None:
        """Log a warning."""
        _format_message("WARN", message, *args)

    @staticmethod
    def error(message: str, *args: object) -> None:
        """Log an error (does not raise; call sys.exit(1) in the caller if needed)."""
        _format_message("ERROR", message, *args)

    @staticmethod
    def json(label: str, value: object) -> None:
        """Log a label and a value; if value is not str, pretty-print as JSON (indent=2)."""
        formatted = json.dumps(value, indent=2, default=str) if not isinstance(value, str) else value
        print(f"[INFO] {label}\n{formatted}")


# Singleton used across the project
logger = _Logger()
